### PR TITLE
New test cases added for order path full flow

### PR DESCRIPTION
**Pull request by JayarathneDha**
Added full order flow test cases with existing deliver address and new deliver address 

Screenshot from 2021-05-04 16-16-36

### CODE REVIEW

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java [hunk 1]
```diff
@@ -1,16 +1,99 @@
...
+  PaymentDetailsPage paymentDetailsPage = new PaymentDetailsPage(getDriver(), 5);
+  OrderConfirmationPage orderConfirmationPage = new OrderConfirmationPage(getDriver(), 5);
+  MyOrder myOrder = new MyOrder(getDriver(), 5);
 
   public CartCheckoutStepDef(Context context) {
     super(context);
   }
+
+  @When("I am click on {string} product")
+  public void i_am_click_on_product(String string) {
```
Comment (darshanalafille) [inline aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:35]: please change to meaningful name

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java [hunk 2]
```diff
@@ -1,16 +1,99 @@
...
+    commonModule.clickOnLinkText(string);
+  }
+
+  @When("I am click on {string}")
+  public void i_am_click_on_link_test(String string) {
```
Comment (darshanalafille) [inline aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:40]: please change to meaningful name

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java [hunk 3]
```diff
@@ -1,16 +1,99 @@
...
+    paymentDetailsPage.enterCardHolderName(cardHolderName);
+    paymentDetailsPage.clickOnExpireMonth();
+    commonModule.clickOnLinkText(expireMonth);
+    paymentDetailsPage.clickOnExpireYear();
+    commonModule.clickOnLinkText(expireYear);
+    paymentDetailsPage.enterCardCod(cardCode);
+  }
+
+  @Then("Verify the {string} Page Header")
+  public void verify_the_page_header(String string) {
```
Comment (darshanalafille) [inline aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:78]: change to meaningful name

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java [hunk 4]
```diff
@@ -1,16 +1,99 @@
...
+    assertThat(orderConfirmationPage.getOrderConfirmationMessage()).containsIgnoringCase(string);
+    context.getOrder().setOrderNumber(orderConfirmationPage.getOrderId().split(":")[1].trim());
+  }
+
+  @Then("I am navigating to {string}")
+  public void i_am_navigating_to(String page) {
```
Comment (darshanalafille) [inline aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:84]: change to meaningful name

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java [hunk 5]
```diff
@@ -1,16 +1,99 @@
...
+    orderConfirmationPage.navigateToPage(PropertyLoader.loadProperty(PropertyLoader.APPLICATION_URL) +"/" + page);
+  }
+
+  @Then("I click on {string} Box")
+  public void i_click_on_box(String string) {
```
Comment (darshanalafille) [inline aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:89]: change to meaningful name

**Review by darshanalafille — CHANGES_REQUESTED** [review]

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java (hunk 1 above)
Comment (JayarathneDha) [reply aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:35]: This is reusable methods. we can use this string value for anything related to relevant step definition. This is a generalize method

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java (hunk 2 above)
Comment (JayarathneDha) [reply aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:40]: This is reusable methods. we can use this string value for anything related to relevant step definition. This is a generalize method

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java (hunk 3 above)
Comment (JayarathneDha) [reply aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:78]: This is reusable methods. we can use this string value for anything related to relevant step definition. This is a generalize method

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java (hunk 4 above)
Comment (JayarathneDha) [reply aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:84]: This is reusable methods. we can use this string value for anything related to relevant step definition. This is a generalize method

File: aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java (hunk 5 above)
Comment (JayarathneDha) [reply aldi-automation/src/test/java/stepdefinitions/aldi/CartCheckoutStepDef.java:89]: This is reusable methods. we can use this string value for anything related to relevant step definition. This is a generalize method

**Review by darshanalafille — APPROVED** [review]

**Pull request merged**